src/Block/MediaBlockService.php

```diff
@@ -151,6 +156,55 @@ public function buildEditForm(FormMapper $formMapper, BlockInterface $block)
         ]);
     }
 
+    /**
+     * NEXT_MAJOR: Remove this method.
+     */
+    public function buildCreateForm(FormMapper $formMapper, BlockInterface $block)
+    {
+        if ('sonata_deprecation_mute' !== (\func_get_args()[2] ?? null)) {
+            @trigger_error(sprintf(
+                'Method %s() is deprecated since sonata-project/media-bundle 3.x and will be removed in version 4.0. Use %s::configureCreateForm() instead.',
+                __METHOD__,
+                static::class
+            ), E_USER_DEPRECATED);
+        }
+        $this->configureCreateForm($formMapper, $block);
+    }
+
+    /**
+     * NEXT_MAJOR: Remove this method.
+     */
+    public function buildEditForm(FormMapper $formMapper, BlockInterface $block)
+    {
+        if ('sonata_deprecation_mute' !== (\func_get_args()[2] ?? null)) {
+            @trigger_error(sprintf(
+                'Method %s() is deprecated since sonata-project/media-bundle 3.x and will be removed in version 4.0. Use %s::configureEditForm() instead.',
+                __METHOD__,
+                static::class
+            ), E_USER_DEPRECATED);
+        }
+        $this->configureEditForm($formMapper, $block);
```

Comment:
FormMapper are not the same:
```php
        public function buildEditForm(Sonata\AdminBundle\Form\FormMapper $form, BlockInterface $block)
    public function configureEditForm(Sonata\BlockBundle\Form\Mapper\FormMapper $form, BlockInterface $block)
```

You should use something like [PageFormMapper](https://github.com/sonata-project/SonataPageBundle/blob/3.x/src/Mapper/PageFormMapper.php):
```php 
$blockMapper = new PageFormMapper($formMapper);
```

Reply:
`Sonata\BlockBundle\Form\Mapper\FormMapper` has been introduced in block bundle 3.13
And this requires 3.17: https://github.com/sonata-project/SonataMediaBundle/blob/3.x/composer.json#L66
block bundles formmapper uses the same adminbundle formmapper without some functions.
Is this really needed?

Reply:
The thing is that when I introduced the BC break in https://github.com/sonata-project/SonataMediaBundle/pull/1699, it was extending `AbstractAdminBlockService`, so to be compatible again, we should just copy [those methods that don't exist in these classes](https://github.com/sonata-project/SonataBlockBundle/blob/3.x/src/Block/Service/AbstractAdminBlockService.php#L43-L83), we cannot change the signature. IMHO the compatibility with BlockBundle 4 should be done in another PR.

Reply:
Yet, somehow these blocks work with my changes. they do not need all of those methods.